README.md
[![Build Status](https://travis-ci.org/soderlind/acf-field-date-time-picker.svg?branch=master)](https://travis-ci.org/soderlind/acf-field-date-time-picker) [![Code Climate](https://codeclimate.com/github/soderlind/acf-field-date-time-picker/badges/gpa.svg)](https://codeclimate.com/github/soderlind/acf-field-date-time-picker)
# ACF  Date and Time Picker Field

Adds a 'Date and Time Picker' field type for the [Advanced Custom Fields](http://wordpress.org/extend/plugins/advanced-custom-fields/) WordPress plugin.

-----------------------

### ACF PRO 5.+ is no longer supported, ACF PRO [has its own date and time picker](https://www.advancedcustomfields.com/resources/date-time-picker/)

### Overview

This is an add-on for the Advanced Custom Fields WordPress plugin, that allows you to add a Date and Time Picker field type.

### Compatibility

This add-on will work with ACF 3 and 4

### Installation


This add-on can be treated as both a WP plugin and a theme include. The plugin is also available from the [WordPress plugin directory](http://wordpress.org/plugins/acf-field-date-time-picker/)

**Install as Plugin**

1. Copy the 'acf-date_time_picker' folder into your plugins folder
2. Activate the plugin via the Plugins admin page

**Include within theme**

1.	Copy the 'acf-date_time_picker' folder into your theme folder (can use sub folders). You can place the folder anywhere inside the 'wp-content' directory
2.	Edit your functions.php file and add the code below (Make sure the path is correct to include the acf-date_time_picker.php file)

```php
add_action('acf/register_fields', 'my_register_fields');

function my_register_fields()
{
	include_once('acf-date_time_picker/acf-date_time_picker.php');
}
```

### Frequently Asked Questions

#### Updating to ACF PRO

I got this quetion over at [AWP on Facebook](https://www.facebook.com/groups/advancedwp/permalink/1198240376904841/?comment_id=1198432300218982&notif_t=group_comment&notif_id=1469033404164280):


>How does upgrading work? If someone starts with ACF and your plugin,
then upgrades to ACF Pro, will their date/time custom field disappear?
I understand the data will be maintained but wondering if the field
will still be visible in WP admin.


Updating to ACF PRO should work fine, ACF PRO has a compatibility add-on for this plugin, but test it on a non production environment first.

Also, read the comments on this issue: [https://github.com/.../acf-field-date-time-picker/issues/103](https://github.com/.../acf-field-date-time-picker/issues/103)



**How do I set the date and time format?**

To set  the date and time format when you create the field, you have to create a string using the letters below.

#### Date format


		d    day of month (no leading zero)
		dd   day of month (two digit)
		o    day of the year (no leading zeros)
		oo   day of the year (three digit)
		D    day name short
		DD   day name long
		m    month of year (no leading zero)
		mm   month of year (two digit)
		M    month name short
		MM   month name long
		y    year (two digit)
		yy   year (four digit)


#### Time format


		H    Hour with no leading 0 (24 hour)
		HH   Hour with leading 0 (24 hour)
		h    Hour with no leading 0 (12 hour)
		hh   Hour with leading 0 (12 hour)
		m    Minute with no leading 0
		mm   Minute with leading 0
		s    Second with no leading 0
		ss   Second with leading 0
		l    Milliseconds always with leading 0
		t    a or p for AM/PM
		T    A or P for AM/PM
		tt   am or pm for AM/PM
		TT   AM or PM for AM/PM


#### Examples

* `yy-mm-dd`: 2013-04-12
* `HH:mm`: 24 hour clock, with a leading 0 for hour and minute
* `h:m tt`: 12 hour clock with am/pm, no leading 0

### More Information

http://soderlind.no/time-picker-field-for-advanced-custom-fields/

### Changelog

Please see [Changelog.md](Changelog.md)


Changelog.md
# Changelog

- [2016-07-20](https://github.com/soderlind/acf-field-date-time-picker/commit/87cc4fa): Update plugin to WPCS standards (committer: soderlind) 
- [2016-07-20](https://github.com/soderlind/acf-field-date-time-picker/commit/fbe7f9a): Update plugin to WPCS standards. Remove ACF PRO support. (committer: soderlind) 
- [2016-07-20](https://github.com/soderlind/acf-field-date-time-picker/commit/32f10b8): Add Travis (committer: soderlind) 
- [2016-07-20](https://github.com/soderlind/acf-field-date-time-picker/commit/8470156): Add (committer: soderlind) 
- [2016-07-20](https://github.com/soderlind/acf-field-date-time-picker/commit/8aa3ccd): Add code climate settings (committer: soderlind) 
- [2016-07-20](https://github.com/soderlind/acf-field-date-time-picker/commit/29bdb65): Added grunt for git and svn deploy (committer: soderlind) 
- [2016-07-20](https://github.com/soderlind/acf-field-date-time-picker/commit/1e8719c): Added wp plugin dir assets (committer: soderlind) 
- [2016-07-20](https://github.com/soderlind/acf-field-date-time-picker/commit/13ce555): Deleted file, ACF PRO is no longer supported (committer: soderlind) 
- [2016-07-20](https://github.com/soderlind/acf-field-date-time-picker/commit/1890870): Renamed files (committer: soderlind) 
- [2016-03-22](https://github.com/soderlind/acf-field-date-time-picker/commit/5368ace): Changed the method of selecting the input elements so that it works on append action (committer: Gareth Chidgey) 
- [2016-03-22](https://github.com/soderlind/acf-field-date-time-picker/commit/5148ff7): Cleaning up JS only, no functional change (committer: Gareth Chidgey) 
- [2016-03-22](https://github.com/soderlind/acf-field-date-time-picker/commit/b514150): Cleaning up JS only, no functional change (committer: Gareth Chidgey) 
- [2016-03-22](https://github.com/soderlind/acf-field-date-time-picker/commit/052210e): Cleaning up PHP, making sure that the path works correctly when symlinked in development, completing missing documentation (committer: Gareth Chidgey) 
- [2015-11-13](https://github.com/soderlind/acf-field-date-time-picker/commit/b83d3f8): add composer.json (committer: Lukas Besch) 
- [2015-01-23](https://github.com/soderlind/acf-field-date-time-picker/commit/3be603f): Fixes Issue #71 (committer: Ryan J. Bonnell) 
- [2014-12-31](https://github.com/soderlind/acf-field-date-time-picker/commit/f19640c): Repeater field compatibility bugfix (committer: jpavlicek) 
- [2014-09-23](https://github.com/soderlind/acf-field-date-time-picker/commit/4a27791): Increase tested version to 4.0 (committer: soderlind) 
- [2014-09-23](https://github.com/soderlind/acf-field-date-time-picker/commit/44babc1): Fixing clone field issue in repeater (committer: soderlind) 
- [2014-09-17](https://github.com/soderlind/acf-field-date-time-picker/commit/30ed1d8): Update timepicker.js (committer: Kamil Grzegorczyk) 
- [2014-09-06](https://github.com/soderlind/acf-field-date-time-picker/commit/7bdd204): Fix Backend Timestamp handling -- tested up WordPress 4.0 (committer: Leo Caseiro) 
- [2014-08-15](https://github.com/soderlind/acf-field-date-time-picker/commit/ff39048): Added link to WordPress plugin directory (committer: soderlind) 
- [2014-08-15](https://github.com/soderlind/acf-field-date-time-picker/commit/44a9ba5): Fix Undefined property: acf_field_date_time_picker::$domain (committer: soderlind) 
- [2014-08-15](https://github.com/soderlind/acf-field-date-time-picker/commit/ea1e488): Link to Changelog.md (committer: soderlind) 
- [2014-08-15](https://github.com/soderlind/acf-field-date-time-picker/commit/cb704e0): Including Changelog.md (committer: soderlind) 
- [2014-08-15](https://github.com/soderlind/acf-field-date-time-picker/commit/7f7bdb0): Added link to Changelog.md (committer: soderlind) 
- [2014-08-15](https://github.com/soderlind/acf-field-date-time-picker/commit/a6a5ba6): Added changelog (committer: soderlind) 
- [2014-08-14](https://github.com/soderlind/acf-field-date-time-picker/commit/2866e28): Fix Undefined property: acf_field_date_time_picker::$domain (committer: Bryan Channon) 
- [2014-08-12](https://github.com/soderlind/acf-field-date-time-picker/commit/88e64b9): problem commiting to svn, had to bump the version number (committer: soderlind) 
- [2014-08-12](https://github.com/soderlind/acf-field-date-time-picker/commit/3a3664b): svn problem, had to bump version (committer: soderlind) 
- [2014-08-12](https://github.com/soderlind/acf-field-date-time-picker/commit/fb05458): the plugin now works with ACF 5.0 (committer: soderlind) 
- [2014-08-12](https://github.com/soderlind/acf-field-date-time-picker/commit/e620121): updated language files (committer: soderlind) 
- [2014-08-05](https://github.com/soderlind/acf-field-date-time-picker/commit/049bfa1): Update README.md (committer: Per Søderlind) 
- [2014-07-28](https://github.com/soderlind/acf-field-date-time-picker/commit/1f02dd6): Fixed a css bug where a regular DatePicker field would be visible if shown with a Date and Timepicker field (committer: Yan Knudtskov Nielsen) 
- [2014-07-21](https://github.com/soderlind/acf-field-date-time-picker/commit/a193ac4): Minor bug fix (committer: Yan Knudtskov Nielsen) 

date_time_picker-v4.php
<?php

class acf_field_date_time_picker extends acf_field
{
    // vars
    var $settings   // will hold info such as dir / path
    , $defaults // will hold default field options
    , $domain   // holds the language domain
    , $lang;

    /*
    *  __construct
    *
    *  Set name / label needed for actions / filters
    *
    *  @since	3.6
    *  @date	23/01/13
    */

    function __construct()
    {
        // vars
        $this->name = 'date_time_picker';
        $this->label = __('Date and Time Picker');
        $this->category = __("jQuery", $this->domain); // Basic, Content, Choice, etc
        $this->domain = 'acf-field-date-time-picker';
        $this->defaults = array(
            'label' => __('Choose Time', $this->domain)
        ,
            'time_format' => 'h:mm tt'
        ,
            'show_date' => 'true'
        ,
            'date_format' => 'm/d/y'
        ,
            'show_week_number' => 'false'
        ,
            'picker' => 'slider'
        ,
            'save_as_timestamp' => 'true'
        ,
            'get_as_timestamp' => 'false'
        );


        // do not delete!
        parent::__construct();


        // settings
        $this->settings = array(
            'path' => apply_filters('acf/helpers/get_path', __FILE__)
        ,
            'dir' => apply_filters('acf/helpers/get_dir', __FILE__)
        ,
            'version' => '2.0.9'
        );

    }

    /*
    *  create_options()
    *
    *  Create extra options for your field. This is rendered when editing a field.
    *  The value of $field['name'] can be used (like bellow) to save extra data to the $field
    *
    *  @type	action
    *  @since	3.6
    *  @date	23/01/13
    *
    *  @param	$field	- an array holding all the field's data
    */

    function create_options($field)
    {
        $field = array_merge($this->defaults, $field);
        $key = $field['name'];
        ?>
        <tr class="field_option field_option_<?php echo $this->name; ?> timepicker_choice">
            <td class="label">
                <label for=""><?php _e("Date and Time Picker?", $this->domain); ?></label>
            </td>
            <td>
                <?php
                do_action('acf/create_field', array(
                    'type' => 'radio'
                ,
                    'name' => 'fields[' . $key . '][show_date]'
                ,
                    'value' => $field['show_date']
                ,
                    'layout' => 'horizontal'
                ,
                    'choices' => array(
                        'true' => __('Date and Time Picker', $this->domain)
                    ,
                        'false' => __('Time Picker', $this->domain)
                    )
                ));
                ?>
            </td>
        </tr>
        <tr class="field_option field_option_<?php echo $this->name; ?> timepicker_dateformat">
            <td class="label">
                <label><?php _e("Date Format", $this->domain); ?></label>
                <p class="description"><?php printf(__("eg. mm/dd/yy. read more about <a href=\"%s\" target=\"_blank\">formatting  date</a>", $this->domain), "http://docs.jquery.com/UI/Datepicker/formatDate"); ?></p>
            </td>
            <td>
                <?php
                do_action('acf/create_field', array(
                    'type' => 'text'
                ,
                    'name' => 'fields[' . $key . '][date_format]'
                ,
                    'value' => $field['date_format']
                ));
                /*
                do_action('acf/create_field', array(
                    'type'	=>	'select',
                    'name'	=>	'fields['.$key.'][date_format]',
                    'value'	=>	$field['date_format'],
                    'choices' => array(
                          'm/d/y'    => 'm/d/y (5/27/13)'
                        , 'mm/dd/yy' => 'mm/dd/yy (05/27/2013)'
                        , 'yy/mm/dd' => 'yy/mm/dd (2013/05/27)'
                        , 'yy-mm-dd' => 'yy-mm-dd (2013-05-27)'
                        , 'dd.mm.yy' => 'dd.mm.yy (27.05.2013)'
                        , 'dd-mm-yy' => 'dd-mm-yy (27-05-2013)'
                        , 'yy-M-dd'  => 'yy-M-dd (2013-May-27)'
                    )
                ));
                */
                ?>
            </td>
        </tr>
        <tr class="field_option field_option_<?php echo $this->name; ?> timepicker_timeformat">
            <td class="label">
                <label><?php _e("Time Format", $this->domain); ?></label>
                <p class="description"><?php printf(__("eg. hh:mm. read more about <a href=\"%s\" target=\"_blank\">formatting  time</a>", $this->domain), "http://trentrichardson.com/examples/timepicker/#tp-formatting"); ?></p>
            </td>
            <td>
                <?php
                do_action('acf/create_field', array(
                    'type' => 'text'
                ,
                    'name' => 'fields[' . $key . '][time_format]'
                ,
                    'value' => $field['time_format']
                ));
                /*
                do_action('acf/create_field', array(
                    'type'	=>	'select',
                    'name'	=>	'fields['.$key.'][time_format]',
                    'value'	=>	$field['time_format'],
                    'choices' => array(
                        'h:mm tt' => 'h:mm tt (9:59 am)'
                        , 'hh:mm tt' => 'hh:mm tt (09:59 am)'
                        , 'H:mm' => 'H:mm (9:59)'
                        , 'HH:mm' => 'HH:mm (09:59)'
                    )
                ));
                */
                ?>
            </td>
        </tr>
        <tr class="field_option field_option_<?php echo $this->name; ?> timepicker_week_number">
            <td class="label">
                <label for=""><?php _e("Display Week Number?", $this->domain); ?></label>
            </td>
            <td>
                <?php
                do_action('acf/create_field', array(
                    'type' => 'radio'
                ,
                    'name' => 'fields[' . $key . '][show_week_number]'
                ,
                    'value' => $field['show_week_number']
                ,
                    'layout' => 'horizontal'
                ,
                    'choices' => array(
                        'true' => __('Yes', $this->domain)
                    ,
                        'false' => __('No', $this->domain)
                    )
                ));
                ?>
            </td>
        </tr>
        <tr class="field_option field_option_<?php echo $this->name; ?> timepicker_week_number">
            <td class="label">
                <label for=""><?php _e("Time Picker style?", $this->domain); ?></label>
            </td>
            <td>
                <?php
                do_action('acf/create_field', array(
                    'type' => 'radio'
                ,
                    'name' => 'fields[' . $key . '][picker]'
                ,
                    'value' => $field['picker']
                ,
                    'layout' => 'horizontal'
                ,
                    'choices' => array(
                        'slider' => __('Slider', $this->domain)
                    ,
                        'select' => __('Dropdown', $this->domain)
                    )
                ));
                ?>
            </td>
        </tr>
        <tr class="field_option field_option_<?php echo $this->name; ?> timepicker_week_number">
            <td class="label">
                <label for=""><?php _e("Save as timestamp?", $this->domain); ?></label>
                <p class="description"><?php printf(__("Most users should leave this untouched, only set it to \"No\" if you need a date and time format not supported by <a href=\"%s\" target=\"_blank\">strtotime</a>", $this->domain), "http://php.net/manual/en/function.strtotime.php"); ?></p>
            </td>
            <td>
                <?php
                do_action('acf/create_field', array(
                    'type' => 'radio'
                ,
                    'name' => 'fields[' . $key . '][save_as_timestamp]'
                ,
                    'value' => $field['save_as_timestamp']
                ,
                    'layout' => 'horizontal'
                ,
                    'choices' => array(
                        'true' => __('Yes', $this->domain)
                    ,
                        'false' => __('No', $this->domain)
                    )
                ));
                ?>
            </td>
        </tr>
        <tr class="field_option field_option_<?php echo $this->name; ?> timepicker_week_number">
            <td class="label">
                <label for=""><?php _e("Get field as timestamp?", $this->domain); ?></label>
                <p class="description"><?php printf(__("Most users should leave this untouched, only set it to \"Yes\" if you need get the  date and time field as a timestamp using  <a href=\"%s\" target=\"_blank\">the_field()</a> or <a href=\"%s\" target=\"_blank\">get_field()</a> ", $this->domain), "http://www.advancedcustomfields.com/resources/functions/the_field/",
                        "http://www.advancedcustomfields.com/resources/functions/get_field/"); ?></p>
            </td>
            <td>
                <?php
                do_action('acf/create_field', array(
                    'type' => 'radio'
                ,
                    'name' => 'fields[' . $key . '][get_as_timestamp]'
                ,
                    'value' => $field['get_as_timestamp']
                ,
                    'layout' => 'horizontal'
                ,
                    'choices' => array(
                        'true' => __('Yes', $this->domain)
                    ,
                        'false' => __('No', $this->domain)
                    )
                ));
                ?>
            </td>
        </tr>
        <?php
    }

    /*
    *  create_field()
    *
    *  Create the HTML interface for your field
    *
    *  @param	$field - an array holding all the field's data
    *
    *  @type	action
    *  @since	3.6
    *  @date	23/01/13
    */

    function create_field($field)
    {

        if ($field['show_date'] !== 'true') {
            echo '<input type="text" value="' . $field['value'] . '" name="' . $field['name'] . '" class="ps_timepicker" value="" data-picker="' . $field['picker'] . '" data-time_format="' . $field['time_format'] . '"  title="' . $field['label'] . '" />';
        } else {
            echo '<input type="text" value="' . $field['value'] . '" name="' . $field['name'] . '" class="ps_timepicker" value="" data-picker="' . $field['picker'] . '" data-date_format="' . $field['date_format'] . '" data-time_format="' . $field['time_format'] . '" data-show_week_number="' . $field['show_week_number'] . '"  title="' . $field['label'] . '" />';
        }
    }

    //function load_field_defaults( $field ) { return $field; }

    function format_value($value, $post_id, $field)
    {
        $field = array_merge($this->defaults, $field);
        if ($value != '' && $field['save_as_timestamp'] == 'true' && $this->isValidTimeStamp($value)) {
            if ($field['show_date'] == 'true') {
                $value = date_i18n(sprintf("%s %s", $this->js_to_php_dateformat($field['date_format']), $this->js_to_php_timeformat($field['time_format'])), $value);
            } else {
                $value = date_i18n(sprintf("%s", $this->js_to_php_timeformat($field['time_format'])), $value);
            }
        }
        return $value;
    }

    function format_value_for_api($value, $post_id, $field)
    {
        $field = array_merge($this->defaults, $field);
        if ($value != '' && $field['save_as_timestamp'] == 'true' && $field['get_as_timestamp'] != 'true' && $this->isValidTimeStamp($value)) {
            if ($field['show_date'] == 'true') {
                $value = date_i18n(sprintf("%s %s", $this->js_to_php_dateformat($field['date_format']), $this->js_to_php_timeformat($field['time_format'])), $value);
            } else {
                $value = date_i18n(sprintf("%s", $this->js_to_php_timeformat($field['time_format'])), $value);
            }
        }
        return $value;
    }

    function js_to_php_dateformat($date_format)
    {
        $chars = array(
            // Day
            'dd' => 'd',
            'd' => 'j',
            'DD' => 'l',
            'D' => 'D',
            'o' => 'z',
            // Month
            'mm' => 'm',
            'm' => 'n',
            'MM' => 'F',
            'M' => 'M',
            // Year
            'yy' => 'Y',
            'y' => 'y',
        );

        return strtr((string)$date_format, $chars);
    }

    function js_to_php_timeformat($time_format)
    {

        $chars = array(
            //hour
            'HH' => 'H',
            'H' => 'G',
            'hh' => 'h',
            'h' => 'g',
            //minute
            'mm' => 'i',
            'm' => 'i',
            //second
            'ss' => 's',
            's' => 's',
            //am/pm
            'TT' => 'A',
            'T' => 'A',
            'tt' => 'a',
            't' => 'a'
        );

        return strtr((string)$time_format, $chars);
    }

    function isValidTimeStamp($timestamp)
    {
        return ((string)(int)$timestamp === (string)$timestamp);
    }

    /*
    *  update_value()
    *
    *  This filter is appied to the $value before it is updated in the db
    *
    *  @type	filter
    *  @since	3.6
    *  @date	23/01/13
    *
    *  @param	$value - the value which will be saved in the database
    *  @param	$post_id - the $post_id of which the value will be saved
    *  @param	$field - the field array holding all the field options
    *
    *  @return	$value - the modified value
    */

    // function update_value( $value, $post_id, $field ) {
    // 	$field = array_merge($this->defaults, $field);
    // 	if ($value != '' && $field['save_as_timestamp'] == 'true') {
    // 		$value = strtotime( $value );
    // 	}
    // 	return $value;
    // }

    function update_value($value, $post_id, $field)
    {
        $field = array_merge($this->defaults, $field);
        if ($value != '' && $field['save_as_timestamp'] == 'true') {
            if (preg_match('/^dd?\//', $field['date_format'])) { //if start with dd/ or d/ (not supported by strtotime())
                $value = str_replace('/', '-', $value);
            }
            $value = strtotime($value);
        }

        return $value;
    }

    /*
    *  input_admin_enqueue_scripts()
    *
    *  This action is called in the admin_enqueue_scripts action on the edit screen where your field is created.
    *  Use this action to add css + javascript to assist your create_field() action.
    *
    *  $info	http://codex.wordpress.org/Plugin_API/Action_Reference/admin_enqueue_scripts
    *  @type	action
    *  @since	3.6
    *  @date	23/01/13
    */

    function input_admin_enqueue_scripts()
    {

        global $wp_locale;

        $has_locale = false;
        $js_locale = $this->get_js_locale(get_locale());

        wp_enqueue_script('jquery-ui-timepicker', $this->settings['dir'] . 'js/jquery-ui-timepicker-addon.js', array(
            'acf-input',
            'jquery-ui-slider'
        ), $this->settings['version'], true);

        if (file_exists($this->settings['path'] . '/js/localization/jquery-ui-timepicker-' . $js_locale . '.js')) {
            wp_enqueue_script('timepicker-localization', $this->settings['dir'] . 'js/localization/jquery-ui-timepicker-' . $js_locale . '.js', array(
                'jquery-ui-timepicker'
            ), $this->settings['version'], true);
            wp_enqueue_script('timepicker', $this->settings['dir'] . 'js/timepicker.js', array(
                'timepicker-localization'
            ), $this->settings['version'], true);
            $has_locale = true;
        } else {
            wp_enqueue_script('timepicker', $this->settings['dir'] . 'js/timepicker.js', array(
                'jquery-ui-timepicker'
            ), $this->settings['version'], true);
        }

        if (!$has_locale && $js_locale != 'en') {
            $timepicker_locale = array(
                'closeText' => __('Done', $this->domain)
            ,
                'currentText' => __('Today', $this->domain)
            ,
                'prevText' => __('Prev', $this->domain)
            ,
                'nextText' => __('Next', $this->domain)
            ,
                'monthStatus' => __('Show a different month', $this->domain)
            ,
                'weekHeader' => __('Wk', $this->domain)
            ,
                'timeText' => __("Time", $this->domain)
            ,
                'hourText' => __("Hour", $this->domain)
            ,
                'minuteText' => __("Minute", $this->domain)
            ,
                'secondText' => __("Second", $this->domain)
            ,
                'millisecText' => __("Millisecond", $this->domain)
            ,
                'timezoneText' => __("Time Zone", $this->domain)
            ,
                'isRTL' => $wp_locale->is_rtl()
            );
        }
        $timepicker_wp_locale = array(
            'monthNames' => $this->strip_array_indices($wp_locale->month)
        ,
            'monthNamesShort' => $this->strip_array_indices($wp_locale->month_abbrev)
        ,
            'dayNames' => $this->strip_array_indices($wp_locale->weekday)
        ,
            'dayNamesShort' => $this->strip_array_indices($wp_locale->weekday_abbrev)
        ,
            'dayNamesMin' => $this->strip_array_indices($wp_locale->weekday_initial)
        ,
            'showMonthAfterYear' => false
        ,
            'showWeek' => false
        ,
            'firstDay' => get_option('start_of_week')
        );

        $l10n = (isset($timepicker_locale)) ? array_merge($timepicker_wp_locale, $timepicker_locale) : $timepicker_wp_locale;
        wp_localize_script('timepicker', 'timepicker_objectL10n', $l10n);

        wp_enqueue_style('jquery-style', $this->settings['dir'] . 'css/jquery-ui.css', array(
            'acf-datepicker'
        ), $this->settings['version']);
        wp_enqueue_style('timepicker', $this->settings['dir'] . 'css/jquery-ui-timepicker-addon.css', array(
            'jquery-style'
        ), $this->settings['version']);
    }

    /**
     * helper function, see: http://www.renegadetechconsulting.com/tutorials/jquery-datepicker-and-wordpress-i18n
     * @param  array $ArrayToStrip
     * @return array
     */
    function strip_array_indices($ArrayToStrip)
    {
        foreach ($ArrayToStrip as $objArrayItem) {
            $NewArray[] = $objArrayItem;
        }

        return $NewArray;
    }

    function get_js_locale($locale)
    {
        $dir_path = $this->settings['path'] . 'js/localization/';
        $exclude_list = array(".", "..");
        $languages = $this->ps_preg_filter("/jquery-ui-timepicker-(.*?)\.js/", "$1", array_diff(scandir($dir_path), $exclude_list));

        $locale = strtolower(str_replace("_", "-", $locale));

        if (false !== strpos($locale, '-')) {
            $l = explode("-", $locale);
            $pattern = array('/' . $locale . '/', '/' . $l[0] . '/', '/' . $l[1] . '/');
        } else {
            $pattern = array('/' . $locale . '/');
        }
        $res = $this->ps_preg_filter($pattern, "$0", $languages, -1, $count);

        return ($count) ? implode("", $res) : 'en';
    }

    function ps_preg_filter($pattern, $replace, $subject, $limit = -1, &$count = 0)
    {
        if (function_exists('preg_filter')) {
            return preg_filter($pattern, $replace, $subject, $limit, $count);
        } else {
            return array_diff(preg_replace($pattern, $replace, $subject, $limit, $count), $subject);
        }
    }
}

// create field
new acf_field_date_time_picker();
